### PR TITLE
Introduce HandleListPage Helper for Streamlined Pagination

### PR DESCRIPTION
**Pull request by fatima2003**
In #678, `path_tidy` was refactored to use pagination. Functions like `doTidyCertStore`, `doTidyRevocationStore`, and `doTidyAcme` each contained loops for processing each cert and page. Other files where pagination will be/ has been introduced will also include these loops.

This PR introduces `HandleListPage()` which is a helper for processing paginated storage lists by supporting item-level (itemCallback) and batch-level (batchCallback) processing. 
- `itemCallback` is invoked for each item in the paginated list, enabling fine-grained control over individual entries. 
- `batchCallback` is invoked after processing a full batch, allowing for cumulative operations on the list.

This abstraction reduces redundancy and enhances consistency across paginated operations.

---

Thank you @cipherboy for the suggestion :)

### CODE REVIEW

**Review by cipherboy — COMMENTED** [review]
Just a minor thing that'd be good to address and then we can merge!

File: changelog/781.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+**Paginated Lists**: Introduce `HandleListPage()` helper to streamline processing of paginated lists using item-level (`itemCallback`) and batch-level (`batchCallback`) callbacks.
```
[the remark this answers is not in the document]
Comment (fatima2003) [reply changelog/781.txt:2]: Makes sense, let me make this change :)

**Comment by fatima2003** [comment]
@cipherboy, it seems `TestTLSValidCert github.com/openbao/openbao/vault/diagnose` and a few others are consistently failing because **vault/diagnose/test-fixtures/goodcertwithroot.pem** is near expiry as seen in this warning which I logged:     
```
/Users/fatima/openbao/vault/diagnose/tls_verification_test.go:36: Warning: 127.0.0.1:443: Leaf certificate 730235215062071302717321348584114824562806123952 is expired or near expiry. Time to expire is: 122h52m32.567212s.
/Users/fatima/openbao/vault/diagnose/tls_verification_test.go:39: warnings returned from good listener
```

I will open up a PR in a couple of hours to fix this or idm if anyone else gets to it first if its an urgent fix :)

**Review by cipherboy — APPROVED** [review]

**Comment by cipherboy** [comment]
Looks great, thank you again @fatima2003!

**Pull request merged**